### PR TITLE
Add featurization of date, timestamp and image columns

### PR DESCRIPTION
**Pull request by imatiach-msft**
Add default featurization for date, timestamp spark types and our internal image type.
For featurization of date columns, convert column to double features:
year
day of week
month
day of month
For featurization of timestamp columns, same as date and in addition:
hour of day
minute of hour
second of minute
For featurization of image columns, use image data converted to double with width and height info

### CODE REVIEW

File: src/featurize/build.sbt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 //> DependsOn: core
 //> DependsOn: utils
 //> DependsOn: multi-column-adapter
+//> DependsOn: readers
```
Comment (drdarshan) [inline src/featurize/build.sbt]: Please avoid adding more dependencies unless really necessary. I wonder if ImageSchema can go into core.

Comment (drdarshan) [inline src/featurize/build.sbt]: Also, I don't think we should handle images implicitly like this..

**Comment by imatiach-msft** [comment]
the dependency on readers was only in the tests, I've removed by generating the images instead of reading them from disk

**Comment by imatiach-msft** [comment]
I've updated the code to only do image featurization for CNTK Learner and disabled it otherwise

**Comment by elibarzilay** [comment]
I have rebased this branch on the current `master`.

**Pull request merged**